src/verify.ts

```diff
@@ -0,0 +1,130 @@
+import * as token from "./token.js"
+import { capabilityCanBeDelegated, DelegationSemantics, DelegationChain, delegationChains, equalCanDelegate, rootIssuer } from "./attenuation.js"
+import { Capability, isCapability } from "./capability/index.js"
+import { Fact, Ucan } from "./types.js"
+
+
+export type Result<Ok, Err = Error>
+  = { ok: true; value: Ok }
+  | { ok: false; error: Err }
+
+const ok: <T, E>(k: T) => Result<T, E> = k => ({ ok: true, value: k })
+const err: <T, E>(e: E) => Result<T, E> = e => ({ ok: false, error: e })
+
+
+/**
+ * Verify a UCAN for an invocation.
+ *
+ * @param ucan
+ *  a UCAN to verify for invocation in JWT format. (starts with 'eyJ...' and has two '.' in it)
+ *
+ * @param audience
+ *  the DID of the callee of this function. The expected audience of the outermost level of the UCAN.
+ *  NOTE: This DID should not be hardcoded in production calls to this function.
+ *
+ * @param isRevoked
+ *  an async predicate on UCANs to figure out whether they've been revoked or not.
+ *  Usually that means checking whether the hash of the UCAN is in a list of revoked UCANs.
```

Comment:
Should we default this to `() => false`?

Reply:
My thinking was "that's too dangerous". But actually defaulting this to not have revocation isn't that dangerous after all. Hm. Maybe we can do that.